undefined reference to 'vtable for Maliit::XCBPlatform'

I want to use the maliit keyboard 2.0 on an embedded Linux system with wayland without X11. Therefore I configure the cmake with 
-Denable-docs=OFF
-Denable-tests=OFF
-Denable-dbus-activation=ON
-Denable-xcb=OFF

[maliit-cmake.txt](https://github.com/maliit/framework/files/7200316/maliit-cmake.txt)

but when I start make I get an error message that there is a undefined reference to 'vtable for Maliit::XCBPlatform'

[maliit-make.txt](https://github.com/maliit/framework/files/7200336/maliit-make.txt)

So what can I do to build maliit without X11 dependencies?

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -194,6 +194,7 @@ if(enable-xcb)
     find_package(XCB REQUIRED xfixes)
     list(APPEND PLUGINS_INCLUDE_DIRS ${XCB_INCLUDE_DIRS})
     list(APPEND PLUGINS_LIBRARIES ${XCB_LIBRARIES})
+    add_definitions(-DHAVE_XCB)
 endif()
 
 if(enable-wayland)

--- a/passthroughserver/main.cpp
+++ b/passthroughserver/main.cpp
@@ -16,7 +16,7 @@
 #include "connectionfactory.h"
 #include "mimserver.h"
 #include "mimserveroptions.h"
-#ifndef NOXCB
+#ifdef HAVE_XCB
 #include "xcbplatform.h"
 #endif
 #ifdef HAVE_WAYLAND

--- a/src/abstractplatform.cpp
+++ b/src/abstractplatform.cpp
@@ -14,7 +14,7 @@
 #include "abstractplatform.h"
 
 #include "unknownplatform.h"
-#ifndef NOXCB
+#ifdef HAVE_XCB
 #include "xcbplatform.h"
 #endif
 #ifdef HAVE_WAYLAND
@@ -40,7 +40,7 @@ std::unique_ptr<AbstractPlatform> createPlatform()
         return std::unique_ptr<AbstractPlatform>(new Maliit::WaylandPlatform);
     }
 #endif
-#ifndef NOXCB
+#ifdef HAVE_XCB
     if (QGuiApplication::platformName() == "xcb") {
         return std::unique_ptr<AbstractPlatform>(new Maliit::XCBPlatform);
     }

